src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -21,48 +25,74 @@
 @WebServlet("/matches")
 public class MatchServlet extends HttpServlet {
 
+  private UserService userService;
+  private NonPersistentUserRepository userRepository;
+
   @Override
   public void init(ServletConfig config) throws ServletException {
     super.init(config);
+    userRepository = new NonPersistentUserRepository();
+    MatchRepository matchRepository = new NonPersistentMatchRepository();
+
+    getServletContext().setAttribute("matchRepository", matchRepository);
+    setLoggedInUser();
+  }
 
-    NonPersistentMatchRepository repository = new NonPersistentMatchRepository();
-    User testUser = repository.addTestData();
+  public void setLoggedInUser() {
+    User currentUser;
+    
+    userService = UserServiceFactory.getUserService();
+    com.google.appengine.api.users.User currentGoogleUser = userService.getCurrentUser();
+    if(currentGoogleUser == null) {
+      currentUser = null;
+    } else {
+      currentUser = userRepository.getUser(currentGoogleUser);
+    }
 
-    //Set MatchRepository and Current User as ServletContext so that they can be accessed by all servlets
-    getServletContext().setAttribute("matchRepository", repository);
-    getServletContext().setAttribute("currentUser", testUser);
+    getServletContext().setAttribute("currentUser", currentUser);
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    setLoggedInUser();
+    Gson gson = new Gson();
     ServletContext servletContext = getServletContext();
-    MatchRepository testRepository = (MatchRepository) servletContext.getAttribute("matchRepository");
-    User testUser = (User) servletContext.getAttribute("currentUser");
+    User currentUser = (User) servletContext.getAttribute("currentUser");
 
-    Collection<User> matches = testRepository.getMatchesForUser(testUser);
+    //is user logged in?
+    if(currentUser == null) {
+        //return null for matches so that page redirects to logged out homepage
+        response.getWriter().println(gson.toJson(null));
+    } else {
+        MatchRepository matchRepository = (MatchRepository) servletContext.getAttribute("matchRepository");
+        Collection<User> matches = matchRepository.getMatchesForUser(currentUser.getId());
 
-    Gson gson = new Gson();
-    response.setContentType("application/json;");
-    response.getWriter().println(gson.toJson(matches));
+        response.setContentType("application/json;");
```

Comment:
Can this (setContentType) be before the if (L63)?
I think you need to set the content type for both branches.

Reply:
Done.